### PR TITLE
Improvement for k8s.io/docs/concepts/services-networking/ingress-controllers/ #31465

### PR DESCRIPTION
**Pull request by PriyanshuAhlawat**
Updated the default backend section for ingress with configuration details.

### CODE REVIEW

**Comment by jihoon-seo** [comment]
@PriyanshuAhlawat Please perform a rebase. 😊

**Comment by mehabhalodiya** [comment]
We couldn't merge this as-is as there is a conflict that the automatic system does not know how to resolve.

@PriyanshuAhlawat  if you haven't done a rebase before, you really have 2 options

1. Learn about Git rebasing. It's a great skill if you often work with source control; on the other hand, if you're mostly a writer or focus your skills in other areas than Git, it's probably not worth the effort (it's genuinely not that easy to learn the first time around).
2. Close this pull request, and open a new one based on the main branch that makes the same changes. In the new PR, mention that it's a replacement for PR #31492 

Here are some references/tips for the following:

- [merge conflicts and rebasing](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#merge-conflicts-and-rebasing)
- [squashing commits](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits)

**Comment by PriyanshuAhlawat** [comment]
@mehabhalodiya  I have resolved the conflict issue and commited  the change please take a look.

**Comment by jimangel** [comment]
@PriyanshuAhlawat it still looks like it could use a squash on the PR. Let us know if you need a hand ([SIG Docs](https://kubernetes.slack.com/messages/sig-docs) on [slack](https://slack.k8s.io/)!)
/lgtm

**Comment by chrismetz09** [comment]
Hi @PriyanshuAhlawat, I know this PR updates the backend. But since you are knowledgeable of ingress controllers and this is the page, it would be great if you could add a really simple/simple diagram. You could even modify the [ingress section diagram](https://kubernetes.io/docs/concepts/services-networking/ingress/) to flesh out the ingress controller component. This makes your PR more valuable to new readers.

Here is the [ingress picture in the mermaid live editor](https://mermaid-js.github.io/mermaid-live-editor/edit/#pako:eNqNkstuwyAQRX8F4U0r2VHqPlSRKqt0UamLqlnaWWAYJygYLB59KMm_Fwsix8qmGwbuXA7DwAEzzQETvDW03yH0_lkrhJgUoNxNFePmtpihN7U1YG3RUUW3wNFLY5ZIaspRQyVVDAyaFUsRXVVybxYDLYlFsTwa7V1YIuMlHC2YL8GgWscYzdY3sRQmvXVgLgAxH80B1mt-V31ovrnWy1EHxeONqLUraFEvqVCoFVKSjHOeW2f0HkjWtm2aF9-Cux156H9ypqU2JJvP54sJZP9sE-K-fGLw-C9KyE0p6YaJNG4lWdM0U0w5YuKJI-ncnTx1IB_6MgzlUOaFLz5mbMBEjlWkGDI4xx2Yjgoe_sVhcNbY7aCDGpMw5dBSL12Na3UKVt9z6uCVC6cNJi2VFnJMvdPrX8UwccbD2bQSNLxtF8XTHzCb3kA).

Hacking simple diagrams using Mermaid is easy. Check out the [Diagram Guide](https://kubernetes.io/docs/contribute/style/diagram-guide/).

**Comment by PriyanshuAhlawat** [comment]
@chrismetz09 thank you for the suggestion , I will try to make the change.

**Comment by chrismetz09** [comment]
@PriyanshuAhlawat, remember keep it simple. Here I took the original ingress picture and changed 2 lines of code so it looks like [this](https://mermaid-js.github.io/mermaid-live-editor/edit/#pako:eNqNkU1vwyAMhv8KIpdNSqou-9BEpV7WHSbtMK3HJAcCpkUlEAHZh6r-9xFBlWa77IIt-_VjYx8xMxwwwTtL-z1Cr--1RogpCdpfVdE218ViUayl3llwrnqJtlmNyl9B9GS0t0YpsE1RrB3YD8mg2kYbS9zQxmZMDc6DvcDEfBSH8t7wm-rN8OZvvJzioHmcmTq3AYF6RaVGQipFMs557sJAByCZECL5xafkfk_u-q-cGWUsyZbL5WoGOTy6hLgtHxjc_4sScnNK-mEiTaUka9t2jiknTOw4kc7bydMG8nEv41OOY17o4rniAmbhOEWyIYNz3IHtqOTh8sdRWWO_hw5qTILLQdBB-RrX-hSkQ8-ph2cuvbGYCKoc5JgO3my_NcPE2wHOoo2k4bZdUp1-AH010-8). Are there other functions (like LB, routing, etc.) that IC provides? Just one simple picture will make it easier for K8s newbies.

Full disclosure: I'm a K8s newbie😉

**Comment by chrismetz09** [comment]
@PriyanshuAhlawat, looked more closely at your PR. Just keep as is so we don't delay merge. I have an action to start hacking on network diagrams.

File: content/en/docs/concepts/services-networking/ingress.md [hunk 1]
```diff
@@ -118,8 +118,11 @@ match a path in the spec.
 
 ### DefaultBackend {#default-backend}
 
-An Ingress with no rules sends all traffic to a single default backend. The `defaultBackend` is conventionally a configuration option
-of the [Ingress controller](/docs/concepts/services-networking/ingress-controllers) and is not specified in your Ingress resources.
+An Ingress with no rules sends all traffic to a single default backend and .spec.defaultBackend is the backend that should handle requests in that case.
```
Comment (tengqm) [inline content/en/docs/concepts/services-networking/ingress.md:121]:
```suggestion
An Ingress with no rules sends all traffic to a single default backend and `.spec.defaultBackend`
is the backend that should handle requests in that case.
```

File: content/en/docs/concepts/services-networking/ingress.md [hunk 2]
```diff
@@ -118,8 +118,11 @@ match a path in the spec.
...
+The `defaultBackend` is conventionally a configuration option of the [Ingress controller](/docs/concepts/services-networking/ingress-controllers) and is not specified in your Ingress resources.
```
Comment (tengqm) [inline content/en/docs/concepts/services-networking/ingress.md:122]:
```suggestion
The `defaultBackend` is conventionally a configuration option of the
[Ingress controller](/docs/concepts/services-networking/ingress-controllers) and
is not specified in your Ingress resources.
```

File: content/en/docs/concepts/services-networking/ingress.md [hunk 3]
```diff
@@ -118,8 +118,11 @@ match a path in the spec.
...
+
+ If no .spec.rules are specified, .spec.defaultBackend must be specified. If defaultBackend is not set, the handling of requests that do not match any of the rules will be up to the
```
Comment (tengqm) [inline content/en/docs/concepts/services-networking/ingress.md:124]:
```suggestion
 If no `.spec.rules` are specified, `.spec.defaultBackend` must be specified.
 If `defaultBackend` is not set, the handling of requests that do not match any of the rules will be up to the
```

**Comment by PriyanshuAhlawat** [comment]
@tengqm can u please take a look at new PR #31831, closed this one as it needed rebase.

**Pull request closed without merging**